silo-vaults/contracts/MetaMorpho.sol

```diff
@@ -738,59 +717,57 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
 
     /// @dev Sets the cap of the market defined by `id` to `supplyCap`.
     /// @dev Assumes that the inputs `marketParams` and `id` match.
-    function _setCap(MarketParams memory marketParams, Id id, uint184 supplyCap) internal {
-        MarketConfig storage marketConfig = config[id];
+    function _setCap(IERC4626 market, uint184 supplyCap) internal virtual {
+        MarketConfig storage marketConfig = config[market];
 
         if (supplyCap > 0) {
             if (!marketConfig.enabled) {
-                withdrawQueue.push(id);
+                withdrawQueue.push(market);
 
                 if (withdrawQueue.length > ConstantsLib.MAX_QUEUE_LENGTH) revert ErrorsLib.MaxQueueLengthExceeded();
 
                 marketConfig.enabled = true;
 
                 // Take into account assets of the new market without applying a fee.
-                _updateLastTotalAssets(lastTotalAssets + MORPHO.expectedSupplyAssets(marketParams, address(this)));
+                _updateLastTotalAssets(lastTotalAssets + _expectedSupplyAssets(market, address(this)));
 
                 emit EventsLib.SetWithdrawQueue(msg.sender, withdrawQueue);
             }
 
             marketConfig.removableAt = 0;
+            IERC20(asset()).approve(address(market), 0);
```

Comment:
In original code, max approval was done on `MORPHO` address. This address works like router. After adjusting this solution for vault standard, we are now calling vaults directly, so each vault needs to be able to pull tokens from `MetaMorpho`. However I need to check deposit method. Our tests usually uses tokens that not need approvals, so this will be small TODO for me.

Reply:
also - my assumption is that MAX approval will be enough for multiple transfers, so we doing just one approval when we set CAP